### PR TITLE
Revert "sim/Make.defs: add -fvisibility=hidden to CFLAGS"

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Revert the change which broke sim on macOS.

## Testing
tested locally
(i had to revert https://github.com/apache/incubator-nuttx/pull/1805 too.)

### CODE REVIEW

File: arch/sim/src/nuttx-names.in [hunk 1]
```diff
@@ -32,57 +32,192 @@
...
 NXSYMBOLS(sigemptyset)
 NXSYMBOLS(sigfillset)
+NXSYMBOLS(sighold)
+NXSYMBOLS(sigismember)
+NXSYMBOLS(sigprocmask)
+NXSYMBOLS(sigrelse)
+NXSYMBOLS(sigtimedwait)
 NXSYMBOLS(sleep)
+NXSYMBOLS(snprintf)
 NXSYMBOLS(socket)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:181]: @yamt, But NuttX's socket is renamed to NXscoket here, we need find why vpnkit_connect call into NXsocket first.

Comment (yamt) [reply arch/sim/src/nuttx-names.in:181]: -unexported_symbols_list is used on macOS. ie. it isn't renamed to NXsocket.

Comment (xiaoxiang781216) [reply arch/sim/src/nuttx-names.in:181]: Ok, we are looking the better approach to fix this issue, please wait a moment.

Comment (yamt) [reply arch/sim/src/nuttx-names.in:181]: sure!

**Comment by xiaoxiang781216** [comment]
Let's close this PR now, since the problem is already fixed by 1817

**Pull request closed without merging**